front-end/src/components/CreatePost.tsx

```diff
@@ -4,26 +4,37 @@ import { Label, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
 import { Post } from '../types/Post';
 import { Form, Input } from 'reactstrap';
 
-export default function CreatePostComponent() {
-    
+const CreatePostComponent = (props: any) => {
+
     const [title, setTitle] = useState('');
     const [desc, setDesc] = useState('');
     const [contentType, setContentType] = useState('');
     const [content, setContent] = useState('');
     const [categories, setCategories] = useState('');
-    const [visibility, setVisibility] = useState('');
-    const [unlisted, setUnlisted] = useState('');
+    const [visibility, setVisibility] = useState('Public');
+    const [unlisted, setUnlisted] = useState(false);
+
+    function changeVisibility(isChecked: boolean) {
+        if (isChecked) {
+            setVisibility("Friends")
+        }
+        else {
+            setVisibility("Public")
+        }
+    }
 
     function sendPost(e:any) {
-        axios.post(process.env.REACT_APP_API_URL + "/api/author/username/posts/", {
+        let data = {
             title: title,
             description: desc,
             visibility: visibility,
             unlisted: unlisted,
             contentType: contentType,
```

Comment:
It might be useful to have contentType as a drop-down, as it reduces potential user error

We could do this at a later point if you'd like (after this PR), but I think it could potentially make demoing go a bit smoother

Reply:
Sounds good, I've changed the contentType to a drop-down :)